test/e2e/utils/chaise.page.js

```diff
@@ -704,7 +704,7 @@ var recordPage = function() {
     };
 
     this.getNoResultsRow = function() {
-        return element(by.id("no-results-row"));
+        return element.all(by.id("no-results-row"));
```

Comment:
Instead of using `all` and getting all "No Results" rows, we should extend this to grab the parent table by it's id, and then get it's corresponding "No Results" row if it's there. This will make the tests more readable.

Reply:
Refactored it.